src/ranging/correction/range_bias/mod.rs

```diff
@@ -0,0 +1,51 @@
+//! Correction of ranging factors based on the following factors:
+//!
+//! * Measured Range
+//! * Channel
+//! * PRF
+
+use core::cmp::{min, max};
+use core::hint::unreachable_unchecked;
+
+// These are the constants
+pub mod ch_1_prf_16;
+pub mod ch_1_prf_64;
+pub mod ch_2_prf_16;
+pub mod ch_2_prf_64;
+pub mod ch_3_prf_16;
+pub mod ch_3_prf_64;
+pub mod ch_4_prf_16;
+pub mod ch_4_prf_64;
+pub mod ch_5_prf_16;
+pub mod ch_5_prf_64;
+pub mod ch_7_prf_16;
+pub mod ch_7_prf_64;
+
+/// Type alias for correction Factors
+pub type CorrectionFactor = [(u16, i16)];
+
+/// Correct a measurement based on range bias.
+///
+/// NOTE: The Correction Factors used must cover the entire range of `measured_range_cm`, e.g.
+/// the last item in the array must have an upper bound of <u16>::max_value(), otherwise this
+/// function may exhibit undefined behavior
+pub fn correct_range_bias(
+    correction: &'static CorrectionFactor,
+    measured_range_cm: u16
+) -> u16 {
+    for (dist, correction) in correction {
+        if measured_range_cm <= *dist {
+            let val = measured_range_cm as i32 + *correction as i32;
+
+            // Clamp to sane u16 values
+            return max(0, min(val, <u16>::max_value() as i32)) as u16;
+        }
+    }
+
+    // NOTE: This is sound if and only if the upper bounds of the correction factor cover the
+    // full u16 range. In this case, it is not possible for the loop not to have already returned
+    // by this point
+    unsafe {
+        unreachable_unchecked();
+    }
```

Comment:
I don't think this is correct. This code is reachable, as the user can pass arbitrary data.

I'd prefer a `panic!` with a descriptive message, and an explanation in the documentation, under which circumstances the function panics.

Reply:
Yeah, this is the tradeoff, I should have been more clear that I wasn't 100% on this decision.

I'd say we can either:

1. Only provide the given CorrectionFactors, and then we can guarantee this is unreachable. The generated code will be slightly smaller (but incorrect if the user provides bad values
2. Be conservative in our loop, but accept arbitrary correction curves.

I think I might take the middle option, and call my current impl something like `correct_range_bias_unchecked`, and maybe make it an unsafe function. I'll implement a copy that has an explicit panic if it reaches that point.

Open to other suggestions as well.

Reply:
Or maybe provide a wrapper type for the correction factors:
``` rust
pub struct CorrectionFactors(&'static [...]);
```
Document the constraint, and provide an `unsafe` constructor. Then the function itself doesn't have to deal with any of it.

Reply:
Good call. I'll look in to this.